pilot/cmd/pilot-agent/config.go

```diff
@@ -33,23 +33,25 @@ import (
 	"istio.io/pkg/log"
 )
 
-func constructProxyConfig() (meshconfig.ProxyConfig, error) {
+// return proxyConfig and trustDomain
+func constructProxyConfig() (meshconfig.ProxyConfig, string, error) {
 	annotations, err := readPodAnnotations()
 	if err != nil {
 		log.Warnf("failed to read pod annotations: %v", err)
 	}
-	var fileMeshContents string
+	var fileMeshContents, trustDomain string
 	if fileExists(meshConfigFile) {
 		contents, err := ioutil.ReadFile(meshConfigFile)
 		if err != nil {
-			return meshconfig.ProxyConfig{}, fmt.Errorf("failed to read mesh config file %v: %v", meshConfigFile, err)
+			return meshconfig.ProxyConfig{}, trustDomain, fmt.Errorf("failed to read mesh config file %v: %v", meshConfigFile, err)
 		}
 		fileMeshContents = string(contents)
 	}
 	meshConfig, err := getMeshConfig(fileMeshContents, annotations[annotation.ProxyConfig.Name])
 	if err != nil {
-		return meshconfig.ProxyConfig{}, err
+		return meshconfig.ProxyConfig{}, trustDomain, err
 	}
+	trustDomain = meshConfig.TrustDomain
```

Comment:
This is not safe. Only gateway has mesh config mounted. Proxies can only rely on ProxyConfig

Reply:
Thanks for the news. But just want to know why the proxy can have different trust-domain.